### PR TITLE
Meson fixes and cleanups

### PR DESCRIPTION
**Pull request by dcbaker**
Initially I started on this because I'm using this in a subproject that set's it's C version to c99, and libpkgconf is full of warnings because of undefined functions. I realized that meson isn't setting a c_std unlike autotools. So i fixed that, added some warnings, updated the CI to be more picky, and then cleaned up the meson a bit (I am a meson dev, I might as well do something useful with all that knowledge :)

### CODE REVIEW

**Comment by kaniini** [comment]
Looks fine, I will merge it into my tree today.  (If I click the button here, it will mess up when my personal git instance mirrors the repo back to Github)

**Comment by dcbaker** [comment]
I updated this PR to add -Werror to the woodpecker CI as well, but otherwise this latest push is exactly the same as the last (since you mentioned it on the other PR)

File: meson.build [hunk 1]
```diff
@@ -110,10 +119,8 @@ pkgconf_exe = executable('pkgconf',
 if get_option('tests')
   kyua_exe = find_program('kyua')
   atf_sh_exe = find_program('atf-sh')
-  test('kyua', kyua_exe, args : ['--config=none', 'test', '--kyuafile=' + join_paths(meson.build_root(), 'Kyuafile'), '--build-root=' + meson.build_root()])
-
-
-  configure_file(input : 'Kyuafile.in', output : 'Kyuafile', configuration : cdata)
+  kyuafile = configure_file(input : 'Kyuafile.in', output : 'Kyuafile', configuration : cdata)
```
Comment (tristan957) [inline meson.build:122]: May want to match the `:` style with the rest of the files. Seem to be attached to the key elsewhere.

Comment (dcbaker) [reply meson.build:122]: The style is all over the place here, and I'm inclined to leave it alone since all I did was assigned the output.

**Comment by dcbaker** [comment]
Migrated to gitea

**Comment by Tachi107** [comment]
> Migrated to gitea

This Gitea migration is kinda confusing to be honest. The README says to report bugs on GitHub, while the mailing list in on sourcehut, where there's also another pkgconf repo.

I guess that bug reporting link in the README simply has to be updated to point to Treehouse's Gitea instance (what's Treehouse btw? It's hard to determine by searching the web), but is the mailing list still monitored? What should it be used for?

**Comment by dcbaker** [comment]
https://gitea.treehouse.systems/ariadne/pkgconf

**Comment by Tachi107** [comment]
> https://gitea.treehouse.systems/ariadne/pkgconf is the url. We really should update the docs.

Yep, already forked that. My main question is about the sourcehut mailing list really (and also who/what Treehouse is)

**Pull request closed without merging**